### ISSUE TITLE
Unrecorded feed language

### ISSUE THREAD

**Issue by StephaneBRT** [PodcastGenerator/PodcastGenerator#384]
**System Information**
Podcast Generator Version: 3.0.1
Webserver: Apache
PHP Version: 7.3

**Podcast Generator Config**
```
$podcastgen_version = "3.0.1"; // Version
$first_installation = xxx;
$installationKey = "xxx";
$scriptlang = "fr_FR";
$url = "https://xxx/";
$absoluteurl = "/xxx/"; // The location on the server
$theme_path = "themes/default/";
$username = "xxx";
$userpassword = "xxx";
$max_upload_form_size = "104857600"; //e.g.: "30000000" (about 30MB)
$upload_dir = "media/"; // "media/" the default folder (Trailing slash required). Set chmod 755
$img_dir = "images/"; // (Trailing slash required). Set chmod 755
$feed_dir = ""; // Where to create feed.xml (empty value = root directory). Set chmod 755
$max_recent = 4; // How many file to show in the home page
$recent_episode_in_feed = "All"; // How many file to show in the XML feed (1,2,5 etc.. or "All")
$episodeperpage = 10;
$enablestreaming = "yes"; // Enable mp3 streaming? ("yes" or "no")
$freebox = "yes"; // enable freely customizable box
$enablepgnewsinadmin = "yes";
$strictfilenamepolicy = "yes"; // strictly rename files (just characters A to Z and numbers) 
$categoriesenabled = "no";
$cronAutoIndex = 1; //Auto Index New Episodes via Cron
$cronAutoRegenerateRSS = 1; //Auto regenerate RSS via Cron
#####################
# XML Feed stuff
$podcast_title = "xxx";
$podcast_subtitle = "xxx";
$podcast_description = "xxx";
$author_name = "xxx";
$author_email = "xxx";
$itunes_category[0] = "xxx"; // iTunes categories (mainCategory:subcategory)
$itunes_category[1] = "xxx";
$itunes_category[2] = "null";
$link = "?name="; // permalink URL of single episode (appears in the <link> and <guid> tags in the feed)
$feed_language = "fr";
$copyright = "Tous droits réservés";   // Your copyright notice (e.g CC-BY)
$feed_encoding = "utf-8";
$explicit = "no"; //does your podcast contain explicit language? ("yes" or "no")
// END OF CONFIG
```
**Description of your problem:**

Hello,
I just installed version 3.0.1 on a shared hosting.
As an administrator, when I modify the details of the podcast and I record, the language of the feed (French) is not kept. It automatically reverts to English.
I have seen a similar issue but I don't see a workable solution.
It's the same for modifying the podcast generator configuration.
The radio buttons :
Activate the audio-video player
Activating the freebox
Activate categories
are on the yes position. When I set them to no and validate, they automatically return to yes.
Could you tell me what to do?
Thank you for your help,
Yours sincerely.

**Comment by Go-rom** [comment]
I can reproduce this with PG 3.1 (clean install)

Meanwhile, you can test this workaround : https://github.com/PodcastGenerator/PodcastGenerator/issues/319

**Comment by Go-rom** [comment]
Your PG installation is out of date.
Run a back up of your current installation and do an upgrade to PG 3.1

**Comment by StephaneBRT** [comment]
I know that version 3.0.1 is not the latest.
But I need the fr language and it is not available on the latest version 3.1. 
So I'm stuck.

**Comment by Go-rom** [comment]
I don't understand how you are stuck. Even if PG is not translated yet in french, you should upgrade because PG 3.1 fixes a lot of bugs.

About your main issue : 
`As an administrator, when I modify the details of the podcast and I record, the language of the feed (French) is not kept. It automatically reverts to English.`

I can tell you this, even if it automatically reverts to English, your choice is kept.
I just ran a test with PG 3.1, i've got the same behaviour but if i look at the config.php file, i see

`$feed_language = "fr";`

So it's only a display bug, the feature works.

If i look at your config file, i see the same ($feed_language = "fr";) so it's ok, your language feed is set to french.

**Comment by StephaneBRT** [comment]
When I did the test on version 3.1 the language of the feed was still in English it seems to me.

And that's not the only problem. I'm talking to French people. The interface has to be in French. So version 3.1 does not suit me until it is translated into French.
So I will wait for the translation.

**Comment by Go-rom** [comment]
3.1 won't be translated in french because it's already released.
We have to wait for @emilengler who will add my translation in the next release (3.2 i think)

Also you are confusing about two features : Feed Language and Interface language.
When you modify "feed language" inside podcast details, it's only to tell that you are producing a french podcast, it's mainly the spoken language used in your podcast.
There's no way in the user interface to change the inferface language. Right now, the only way is to modify your config.php with the variable : "$scriptlang". It's currently useless because as you're noticing, french translation doesn't exist yet inside PG.

French translation is ready (see : https://www.transifex.com/albertobeta/podcast-generator/language/fr_FR/), just wait the next release.
About upgrading to 3.1, you do as you want but know that PG 3.0.1 is not supported anymore.

For now, you should close this issue.

**Comment by Go-rom** [comment]
Hello @StephaneBRT 
Just let you know that i've made some progresses on the french translation. Now, i'm able to test it, some adjustments are needed right now. Next step is to produce a new version. If you want to test the fixed version later, please tell me.

**Comment by emilengler** [comment]
Hello @StephaneBRT sorry for my long time!
Do you have shell access to the server? If so, can you post the output of `locale -a`?

**Comment by chiliec** [comment]
Have the same issue with ru_RU language. Don't changes from admin and don't working when changing from config.

**Comment by JolleTC** [comment]
I'm using version 3.1.1 and tried to add my podcast to iTunes. But in iTunes i get the information that the language is not set in my RSS. 
Though in my feed.xml there is a line with "<language>de</language>". I changed nothing, that's how it was generated with this tool.
Can anyone help me?

**Comment by coldacid** [comment]
French will be one of the included languages in 3.2 (coming soon!) and has a completed translation.
